src/Monolog/Logger.php

```diff
@@ -311,6 +308,11 @@ public function addRecord(int $level, string $message, array $context = []): boo
             $record = call_user_func($processor, $record);
         }
 
+        reset($this->handlers);
+        while ($key !== key($this->handlers)) {
+            next($this->handlers);
+        }
+
         while ($handler = current($this->handlers)) {
```

Comment:
@staabm it would, if it was an easy way to rewind to the place we stopped in the first loop. My first implementation was `foreach`-based, but it looked uglier. Additionally, the minor overhead of `next/current` would be amortized by generally heavy handlers.